lms/models/event.py

```diff
@@ -0,0 +1,113 @@
+from enum import Enum
+
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict
+
+from lms.db import BASE, varchar_enum
+
```

Comment:
Everything this going to be very normalized here

The thinking is to keep the base table very small, allowing the time filter queries to be relatively fast and then pull any extra data as needed.